eth/consensus/pow.py

```diff
@@ -30,23 +31,33 @@
 
 
 # Type annotation here is to ensure we don't accidentally use strings instead of bytes.
-cache_seeds = [b'\x00' * 32]  # type: List[bytes]
+cache_seeds = {0: b'\x00' * 32}  # type: Dict[int, bytes]
 cache_by_seed = OrderedDict()  # type: OrderedDict[bytes, bytearray]
 CACHE_MAX_ITEMS = 10
 
 
 def get_cache(block_number: int) -> bytes:
-    while len(cache_seeds) <= block_number // EPOCH_LENGTH:
-        cache_seeds.append(keccak(cache_seeds[-1]))
-    seed = cache_seeds[block_number // EPOCH_LENGTH]
+    # Generate the seed for the cache if it's missing, aim for thread-safety
+    epoch_index = block_number // EPOCH_LENGTH
+    while epoch_index not in cache_seeds:
+        nearest_index = min(epoch_index - 1, max(cache_seeds.keys()))
```

Comment:
This `min` check seems totally unnecessary, I think `nearest_index = max(cache_seeds.keys())` is sufficient.